### PR TITLE
Seq stats

### PR DESCRIPTION
**Pull request by dfsp-spirit**
This allows Syncopy Developers (not users) to register an additonal HDF5 dataset on a SyncopyData object.
These extra dataset are **not** intended to be used in any parallel operations. They are intended to be used for things like sequential statistics, which can be computed on a single machine/core.

Author Guidelines
-----------------
- [x] Is the change set **< 600 lines**?
- [x] Was the code checked for memory leaks/performance bottlenecks?
- [x] Is the code running locally **and** on the ESI cluster?
- [x] Is the code running on all supported platforms?

Reviewer Checklist
------------------
- [ ] Are testing routines present?
- [ ] Do **parallel** loops have a set length and correct termination conditions?
- [ ] Do objects in the global package namespace perform proper parsing of their input? 
- [ ] Do code-blocks provide novel functionality, i.e., no re-factoring using builtin/external packages possible?
- [ ] Code layout
  - [ ] Is the code PEP8 compliant?
  - [ ] Does the code adhere to agreed-upon naming conventions?
  - [ ] Are keywords clearly named and easy to understand?
  - [ ] No commented-out code?
- [ ] Are all docstrings complete and accurate?
- [ ] Is the CHANGELOG.md up to date?

### CODE REVIEW

File: syncopy/datatype/continuous_data.py [hunk 1]
```diff
@@ -724,9 +746,19 @@ class TimeLockData(ContinuousData):
     """
 
     _infoFileProperties = ContinuousData._infoFileProperties
+    _hdfFileDatasetProperties = ContinuousData._hdfFileDatasetProperties + ("avg", "var", "cov",)
     _defaultDimord = ["time", "channel"]
     _stackingDimLabel = "time"
 
+    # A `h5py.Dataset` holding the average of `data`, or `None` if not computed yet.
+    _avg = None
```
Comment (tensionhead) [inline syncopy/datatype/continuous_data.py:754]: Do we really need class attributes here?

File: syncopy/tests/test_attach_dataset.py [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+        """
+        def some_local_func():
+            adt = _get_fooof_signal()
+            assert isinstance(adt, spy.AnalogData)
+
+            # Copying
+            adt2 = adt.copy()
+
+            extra_data = np.zeros((3, 3), dtype=np.float64)
+            adt._register_seq_dataset("dset_mean", extra_data)
```
Comment (tensionhead) [inline syncopy/tests/test_attach_dataset.py:176]: That's a very nice interface imho :+1:

File: syncopy/datatype/base_data.py [hunk 3]
```diff
@@ -155,7 +169,71 @@ def container(self):
...
+                act = "read-only file"
+                raise SPYValueError(legal=lgl, varname="mode", actual=act)
+            if isinstance(self._data, h5py.Dataset):
+                if isinstance(self._data.file, h5py.File):
+                    if propertyName in self._data.file.keys():
+                        del self._data.file[propertyName]
+                else:
+                    SPYWarning("Could not delete dataset from file.")
+
+    def _update_seq_dataset(self, propertyName, inData=None):
```
Comment (tensionhead) [inline syncopy/datatype/base_data.py:230]: Here the elegance of the register/unregister interface shows :slightly_smiling_face:

File: syncopy/datatype/base_data.py [hunk 4]
```diff
@@ -589,23 +703,13 @@ def sampleinfo(self):
 
     @sampleinfo.setter
     def sampleinfo(self, sinfo):
-        raise SPYError("Cannot set sampleinfo. Use `BaseData._trialdefinition` instead.")
-
-    @property
-    def trialintervals(self):
```
Comment (tensionhead) [inline syncopy/datatype/base_data.py:595]: Wait, it's not good to remove that guy?!

**Review by tensionhead — COMMENTED** [review]
Thanks @dfsp-spirit This is a huge step forward for Syncopy, and will allow more complex data structures immediately needed for timlock analysis :+1: 

Not sure what's happening with the diff, if you could do one more sanity check.. otherwise ready to merge :slightly_smiling_face:

**Comment by dfsp-spirit** [comment]
Closing this due to ugly history.

**Comment by dfsp-spirit** [comment]
I closed this due to the ugly, force-pushed history. I have re-opened a new PR without the force-push.

**Pull request closed without merging**